tool/src/org/antlr/v4/codegen/model/Lexer.java

```diff
@@ -28,7 +29,20 @@ public Lexer(OutputModelFactory factory, LexerFile file) {
 		this.file = file; // who contains us?
 
 		Grammar g = factory.getGrammar();
-		channels = new LinkedHashMap<String, Integer>(g.channelNameToValueMap);
+		Target target = factory.getGenerator().getTarget();
+
+		channels = new LinkedHashMap<>();
+		channelNames = new ArrayList<>();
```

Comment:
Shouldn't this be escaped channel names or something?  seems inconsistent

Reply:
Yes, inconsistent. More correctly: `orginialChannelNames` or `runtimeChannels` but I wanted to introduce minimal code changes.

Reply:
Ok, let's go with consistency and more extensive changes here. thanks!